src/libs/memoize/stats.ts

```diff
@@ -0,0 +1,105 @@
+type MemoizeStatsEntry = {
+    keyLength: number;
+    didHit: boolean;
+    cacheRetrievalTime: number;
+    fnTime?: number;
+    cacheSize: number;
+};
+
+class MemoizeStats {
+    private calls = 0;
+
+    private hits = 0;
+
+    private avgKeyLength = 0;
+
+    private avgCacheRetrievalTime = 0;
+
+    private avgFnTime = 0;
+
+    private cacheSize = 0;
+
+    enabled = false;
+
+    constructor(enabled: boolean) {
+        this.enabled = enabled;
+    }
+
+    // See https://en.wikipedia.org/wiki/Moving_average#Cumulative_average
+    private calculateCumulativeAvg(avg: number, length: number, value: number) {
+        const result = (avg * (length - 1) + value) / length;
+        return Number.isFinite(result) ? result : avg;
+    }
+
+    private cumulateEntry(entry: MemoizeStatsEntry) {
+        this.calls++;
+        this.hits += entry.didHit ? 1 : 0;
+
+        this.cacheSize = entry.cacheSize;
+
+        this.avgKeyLength = this.calculateCumulativeAvg(this.avgKeyLength, this.calls, entry.keyLength);
+
+        this.avgCacheRetrievalTime = this.calculateCumulativeAvg(this.avgCacheRetrievalTime, this.hits, entry.cacheRetrievalTime);
+
+        if (entry.fnTime !== undefined) {
+            this.avgFnTime = this.calculateCumulativeAvg(this.avgFnTime, this.calls - this.hits, entry.fnTime);
+        }
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    static isMemoizeStatsEntry(entry: any): entry is MemoizeStatsEntry {
```

Comment:
This is only used internally so it's unclear why it would be static instead of private

Reply:
Hoisted it out of the class as a standalone identity function. It will save "some bytes" comparing to having instance of this function every time class is instantiated.